calico_node/filesystem/startup.py

```diff
@@ -259,9 +260,10 @@ def main():
     # Write a startup environment file containing the IP address that may have
     # just been detected.
     # This is required because the confd templates expect to be able to fill in
-    #  the IP by fetching it from the environment.
+    # some templates by fetching them from the environment.
     with open('startup.env', 'w') as f:
-        f.write("IP=" + ip)
+        f.write("export IP=%s\n" % ip)
```

Comment:
I just ran a `calicoctl node --ip=<not default selection>` with a recent calico-node and confd picked up the IP variable fine.

What makes you say "this just didn't work before"?